common/OpTestInstallUtil.py

```diff
@@ -453,15 +455,23 @@ def update_kernel_cmdline(self, args="", remove_args="", reboot=True):
                     cmd = "sed -i 's/%s=.*/%s=\"%s\"/g' %s" % (grub_key, grub_key,
                                                                output, boot_cfg)
                     con.run_command(cmd, timeout=60)
-                    con.run_command("update-grub")
+                    if 'Ubuntu' in distro:
+                        con.run_command('update-grub')
+                    else:
+                        con.run_command('grub2-mkconfig -o /boot/grub2/grub.cfg')
                 except CommandFailed as Err:
                     print(("Failed to update kernel commandline - %s: %s" %
                            (Err.command, Err.output)))
                     return False
         if reboot and (req_args or req_remove_args):
             # Reboot the host for the kernel command to reflect
-            self.cv_SYSTEM.goto_state(OpSystemState.OFF)
-            self.cv_SYSTEM.goto_state(OpSystemState.OS)
+            if reboot_cmd:
+                raw_pty = self.cv_SYSTEM.console.get_console()
```

Comment:
Can we use reboot parameter itself to decide a hard / soft reboot. Instead of boolean reboot, let us have reboot variable to hold either "hard" or "soft" Default as "hard", this can solve the purpose and we can avoid an additional variable reboot_cmd and a check.

Reply:
Please do paste the test logs too.

Reply:
> Can we use reboot parameter itself to decide a hard / soft reboot. Instead of boolean reboot, let us have reboot variable to hold either "hard" or "soft" Default as "hard", this can solve the purpose and we can avoid an additional variable reboot_cmd and a check.

If we use that, then there is no way to use the function without reboot. i.e `reboot=False` which is required by other tests. With `reboot="hard"` it will always be set to true. Also if the type of the reboot variable is changed the tests using that also needs to be changed.

Reply:
> Please do paste the test logs too.

Logs are posted here
soft reboot: [use_reboot.log](https://github.com/open-power/op-test/files/6701475/use_reboot.log)

hard reboot: [without_reboot.log](https://github.com/open-power/op-test/files/6701476/without_reboot.log)

Reply:
> Please do paste the test logs too.

Logs are posted here
soft reboot: [use_reboot.log](https://github.com/open-power/op-test/files/6701498/use_reboot.log)

hard reboot: [hard_reboot.log](https://github.com/open-power/op-test/files/6701500/hard_reboot.log)